spec/services/notifications/milestone/send_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "rails_helper"
+
+RSpec.describe Notifications::Milestone::Send do
+  let(:user) { create(:user) }
+  let(:article) { create(:article, user_id: user.id, page_views_count: 4000, positive_reactions_count: 70) }
+  let(:view_milestone_hash) { { type: "View", article: article } }
+  let(:reaction_milestone_hash) { { type: "Reaction", article: article } }
+
+  context "when a user has never received a milestone notification" do
+    it "sends the appropriate level view milestone notification" do
+      send_milestone_notification_view
+      expect(user.notifications.first.action).to include "2048"
+    end
+
+    it "sends the appropriate level reaction milestone notification" do
+      send_milestone_notification_reaction
+      expect(user.notifications.first.action).to include "64"
+    end
+  end
+
+  context "when a user has received a milestone notification before" do
+    def mock_previous_view_milestone_notification
+      send_milestone_notification_view
+      article.update_column(:page_views_count, 9001)
+      send_milestone_notification_view
+    end
+
+    def mock_previous_reaction_milestone_notification
+      send_milestone_notification_reaction
+      article.update_column(:positive_reactions_count, 150)
+      send_milestone_notification_reaction
+    end
+
+    it "sends the appropriate level view milestone notification" do
+      mock_previous_view_milestone_notification
+      expect(user.notifications.second.action).to include "8192"
```

Comment:
Probably these tests could check the whole `action` values.

Reply:
Here include is meant to check a String, and not an Array. The second.action is like "View::8192".
And the first.action is checked elsewhere.